terraform/state.go

```diff
@@ -677,11 +673,13 @@ func (m *ModuleState) deepcopy() *ModuleState {
 		return nil
 	}
 	n := &ModuleState{
-		Path:      make([]string, len(m.Path)),
-		Outputs:   make(map[string]interface{}, len(m.Outputs)),
-		Resources: make(map[string]*ResourceState, len(m.Resources)),
+		Path:         make([]string, len(m.Path)),
+		Outputs:      make(map[string]interface{}, len(m.Outputs)),
+		Resources:    make(map[string]*ResourceState, len(m.Resources)),
+		Dependencies: make([]string, len(m.Dependencies)),
 	}
 	copy(n.Path, m.Path)
+	copy(n.Dependencies, m.Dependencies)
```

Comment:
Yep actually just a bug that we never copied those over in the copy function and we needed them for the mv.